ste/xferRetryHelper.go

```diff
@@ -37,17 +37,19 @@ func getShouldRetry() func(*http.Response, error) bool {
 		return nil
 	}
 	return func(resp *http.Response, err error) bool {
-		if storageErrorCodes, ok := RetryStatusCodes[resp.StatusCode]; ok {
-			// no status codes specified to compare to
-			if len(storageErrorCodes) == 0 {
-				return true
-			}
-			// compare to status codes
-			errorCode := getErrorCode(resp)
-			if errorCode != "" {
-				if _, ok = storageErrorCodes[errorCode]; ok {
+		if resp != nil {
+			if storageErrorCodes, ok := RetryStatusCodes[resp.StatusCode]; ok {
```

Comment:
Should we put explicit logs here saying we are retrying due to custom policy etc.

Reply:
Sure, I think that's a good idea. Let me do that.